src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate process-csr-deletion!
+  :- {:outcome (schema/enum :success :not-found :error) :message schema/Str}
+  "Delete pending certificate requests for subject"
+  [subject :- schema/Str
+   {:keys [csrdir] :as settings} :- CaSettings]
```

Comment:
Just FYI - sometimes I leave the `:as` part even if it's unused, because it can be useful as documentation to the next developer when they are wondering what is passed in.  Not always necessary, but sometimes it is useful.  (In this case I think you are fine either way.)